### PR TITLE
Migrate from yarn to npm + node upgrade

### PR DESCRIPTION
**Pull request by chrmod**

### CODE REVIEW

File: webpack.config.js [hunk 1]
```diff
@@ -17,6 +17,11 @@ const webpack = require('webpack');
 const MiniCssExtractPlugin = require('mini-css-extract-plugin');
 const WebpackShellPlugin = require('webpack-shell-plugin');
 const { CleanWebpackPlugin } = require('clean-webpack-plugin');
+const sass = require('sass');
+const crypto = require('crypto');
+
+const crypto_orig_createHash = crypto.createHash;
+crypto.createHash = algorithm => crypto_orig_createHash(algorithm == 'md4' ? 'sha256' : algorithm);
```
Comment (philipp-classen) [inline webpack.config.js:24]: md4 or should it be rather md5? Do you remember why this was needed?

File: package.json [hunk 2]
```diff
@@ -128,5 +127,8 @@
     "webpack": "^4.46.0",
     "webpack-cli": "^3.3.12",
     "webpack-shell-plugin": "^0.5.0"
+  },
+  "engines": {
+    "yarn": "YARN NO LONGER USED - use npm instead."
```
Comment (smalluban) [inline package.json:132]: is this a good practice, when the engine is going to be deprecated? I suppose that this field should take a version number, not a warning massage.

File: webpack.config.js (hunk 1 above)
Comment (smalluban) [inline webpack.config.js:24]: Linter does not scan this file, but my editor suggests to change `==` to `===` ;)

**Review by smalluban — CHANGES_REQUESTED** [review]
* `yarn.lock` leftover
* Not sure about the `engine` field
* some simple clean up
* No idea why we monkey patch the crypto

Nevertheless, I was able to build the project and run it in Chrome. I did not notice CSS issues (but the mixin is in the android folder, so I suppose that those styles are only applied on that platform).

File: webpack.config.js (hunk 1 above)
Comment (chrmod) [reply webpack.config.js:24]: This is sadly something that I don't have a grasp on. But in short, webpack uses some hash functions that are no longer provided by the crypto package. A proper fix would be to update webpack, but that solution requires way more effort.

Comment (chrmod) [reply webpack.config.js]: I've added some comments to make it clear

**Review by philipp-classen — DISMISSED** [review]

**Pull request merged**